mesonbuild/modules/python.py

```diff
@@ -247,6 +247,7 @@ def wrap_in_pythons_pc_dir(name: str, env: 'Environment', kwargs: T.Dict[str, T.
             try:
                 return PythonPkgConfigDependency(name, env, kwargs, installation)
             finally:
+                os.environ.pop('PKG_CONFIG_LIBDIR', None)
```

Comment:
I really dislike using pop here since we're not taking the valuie, and pop is really more about move than deleting I guess we could write as:
```python
if old_pkg_libdir is not None:
  os.environ['PKG_CONFIG_LIBDIR'] = old_pkg_libdir
elif pkg_libdir:
  del os.environ['PKG_CONFIG_LIBDIR']
```

Really though this manual cleanup is really a pain, and I wish os.environ just has a "with os.environ.temp({'foo': 'bar'}:` that would just do the cleanup automatically and not really that nice.

Reply:
ok, removed usage of pop here.